src/planner/planner.c

```diff
@@ -160,12 +160,24 @@ static void
 rte_mark_for_expansion(RangeTblEntry *rte)
 {
 	Assert(rte->rtekind == RTE_RELATION);
-	Assert(rte->ctename == NULL || rte->ctename == TS_FK_EXPAND);
-	if (rte->ctename != TS_FK_EXPAND)
-		rte->ctename = (char *) TS_CTE_EXPAND;
+	Assert(rte->ctename == NULL);
+	rte->ctename = (char *) TS_CTE_EXPAND;
 	rte->inh = false;
 }
 
+static void
+rte_mark_for_fk_expansion(RangeTblEntry *rte)
+{
+	Assert(rte->rtekind == RTE_RELATION);
+	Assert(rte->ctename == NULL);
+	rte->ctename = (char *) TS_FK_EXPAND;
+	/*
+	 * If this is for an FK lookup query inherit should be false
+	 * initially for hypertables.
+	 */
+	Assert(!rte->inh);
+}
+
 bool
```

Comment:
Maybe we should delete the final strcmp here while we're at it? We always use pointer equality so that case looks weird.

Reply:
in `ts_rte_is_marked_for_expansion` I mean

Reply:
almost removed it, but hesitated, but I think its a good idea

Reply:
done

Reply:
ok had to revert that cause it was causing test failures